px4_ros2_cpp/include/px4_ros2/utils/geometry.hpp

```diff
@@ -181,12 +171,7 @@ Type quaternionToPitch(const Eigen::Quaternion<Type> & q)
 template<typename Type>
 Type quaternionToYaw(const Eigen::Quaternion<Type> & q)
 {
-  Type x = q.x();
-  Type y = q.y();
-  Type z = q.z();
-  Type w = q.w();
-
-  return std::atan2(2.0 * (x * y + w * z), w * w + x * x - y * y - z * z);
+  return quaternionToEulerRpy(q)[0];
```

Comment:
```suggestion
  return quaternionToEulerRpy(q)[2];
```
This method returns the yaw

Reply:
typo, sorry about that